### PR TITLE
Replacing SCSS with styled components

### PR DESCRIPTION
**Pull request by Shriever**
Just made my first commit so I wanted to open a PR to keep track of changes and open a discussion. 

This PR is for the following issue - #2261

### CODE REVIEW

**Comment by Shriever** [comment]
## Issue
On commit [970e42c](https://github.com/tmrowco/electricitymap-contrib/pull/3335/commits/970e42ccf64060f561f8e55dd62dcabec8b039eb), `docker-compose build` stops working with this message in the console `error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
The command '/bin/sh -c yarn lint' returned a non-zero code: 1
ERROR: Service 'web' failed to build : Build failed`

## More details
Before my first commit it worked fine and became a bit slower after [2a8455c](https://github.com/tmrowco/electricitymap-contrib/pull/3335/commits/2a8455c3d0b44848156bca0f4931ae406cdd43e7). By the second commit, it stopped working altogether.

## What I tried
I couldn't find any glaring bugs in the code or the console when running `docker-compose up`. I'm fairly new to Docker so I may have missed something obvious. However, it seems to have something to do with the linting.

**Comment by Shriever** [comment]
## Potential solution
On circleci, I found that location is being set with the `useLocation` hook and my styles are outside of the component. I moved the styled components that needed to access `location` into the component and now `docker-compose build` runs properly.

## Another issue
According to the console, it is not recommended to dynamically call `styled` inside of a react component. Maybe there is some way to access location without the `useLocation` hook? If so, we would be able to move the styled component declarations outside of the component.

**Comment by skovhus** [comment]
Does running `yarn lint` locally work?

**Comment by Shriever** [comment]
I'm having trouble getting the deps installed locally, but it seems like the circleci lint tests are passing. I think that issue was just because `location` which is returned from the `useLocation` hook, was declared and never used.

**Comment by skovhus** [comment]
> I'm having trouble getting the deps installed locally

Ah okay, that would be nice to fix for future contributors. Can you add information about what fails when you run `yarn install` locally and add your version of node.js? 

Thanks. :)

File: web/src/layout/leftpanel/index.js [hunk 1]
```diff
@@ -60,19 +60,30 @@ const mapStateToProps = state => ({
   isLeftPanelCollapsed: state.application.isLeftPanelCollapsed,
 });
 
+// Hide the panel completely if looking at the map on small screens.
+const Container = styled.div`
+  @media (max-width: 767px) {
+    display: ${location.pathname === '/map' ?  'none !important': 'flex'};
```
[the remark this answers is not in the document]
Comment (Shriever) [reply web/src/layout/leftpanel/index.js]: Nice! That got rid of the errors in the console. 
I didn't know you could access props like that in a styled component.

**Comment by Shriever** [comment]
Node Version - `v14.17.0`

***I should mention that I am not running yarn on my machine.**

After running `npm i` in `electricitymap-contrib/web` the process ended with the following error message: 
```npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /home/levi/open-source/electricitymap-contrib/web/node_modules/topojson/node_modules/topojson-server/bin/geo2topo
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/home/levi/open-source/electricitymap-contrib/web/node_modules/topojson/node_modules/topojson-server/bin/geo2topo'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/levi/.npm/_logs/2021-08-30T12_45_02_451Z-debug.log
```

Seems like I'm missing a file. I would guess that I'm missing something because my `npm i` doesn't access the yarn.lock file.

**Comment by madsnedergaard** [comment]
> Node Version - `v14.17.0`
> 
> ***I should mention that I am not running yarn on my machine.**
> 
> After running `npm i` in `electricitymap-contrib/web` the process ended with the following error message:
> 
> ```
> npm ERR! syscall chmod
> npm ERR! path /home/levi/open-source/electricitymap-contrib/web/node_modules/topojson/node_modules/topojson-server/bin/geo2topo
> npm ERR! errno -2
> npm ERR! enoent ENOENT: no such file or directory, chmod '/home/levi/open-source/electricitymap-contrib/web/node_modules/topojson/node_modules/topojson-server/bin/geo2topo'
> npm ERR! enoent This is related to npm not being able to find a file.
> npm ERR! enoent 
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/levi/.npm/_logs/2021-08-30T12_45_02_451Z-debug.log
> ```
> 
> Seems like I'm missing a file. I would guess that I'm missing something because my `npm i` doesn't access the yarn.lock file.

Could you try with `yarn` instead of `npm`? :)

**Comment by Shriever** [comment]
Got yarn on my machine! I have to say that the logs are much cleaner than npm.

Looks like it works fine. I got the deps installed and ran `yarn server-dev` without errors. I did have to change the port number for the server since I don't have a `.env` file.

**Comment by Shriever** [comment]
Hey all, it's been a few days since the last update on this PR. Just wanted to jump in and ask if anything else needs doing before we can merge? Thanks.

**Comment by skovhus** [comment]
Hi! Thanks for fixing this. Have you already done local testing of all the changes and verified that it looks and behaves the same?

File: web/src/components/tooltips/mapcountrytooltip.js [hunk 2]
```diff
@@ -12,9 +13,16 @@ const mapStateToProps = state => ({
...
+
 const TooltipContent = React.memo(({
   isDataDelayed, hasParser, co2intensity, fossilFuelPercentage, renewablePercentage,
 }) => {
+  const CountryTableHeaderInner = styled.div`
+    display: flex;
+    flex-basis: 33.3%;
+    justify-content: space-between;
+  `;
+
```
Comment (madsnedergaard) [inline web/src/components/tooltips/mapcountrytooltip.js:25]: This styled component must be created outside of `TooltipContent`:

Screenshot 2021-09-09 at 09 03 06

**Review by madsnedergaard — CHANGES_REQUESTED** [review]
Great work here and thanks for taking the time to do this - this code is quite messy and it's a lot of work, so it's really appreciated! 💪 
Two comments, otherwise it's all good :)

**Comment by Shriever** [comment]
Took care of those comments. While I was working on it I checked that the styles were still being applied and all looked good.
Could be that I missed something, but I think the functionality remained the same.

File: web/src/layout/leftpanel/zonedetailspanel.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ const mapStateToProps = state => ({
 
 const SocialButtons = styled.div`
   @media (max-width: 767px) {
-    display: ${location.pathname !== '/map' ? 'none !important' : 'block' };
+    display: ${props => props.pathname !== '/map' ? 'none !important' : 'block' };
```
Comment (madsnedergaard) [inline web/src/layout/leftpanel/zonedetailspanel.js:29]: You need to also add this `pathname` prop to `<SocialButtons />` :)

Comment (Shriever) [reply web/src/layout/leftpanel/zonedetailspanel.js]: Sorry about that one. Didn't notice.

**Comment by Shriever** [comment]
Props have been passed!

**Review by madsnedergaard — APPROVED** [review]
Wohooo, awesome! 🥳

**Comment by Shriever** [comment]
Yay! This was my first code contribution. Thanks for your help with this!

**Pull request merged**